### DISCUSSION TITLE
Alert/Automation for Crashes/No Frames Received?

### DISCUSSION THREAD

**Discussion by matknowles** [blakeblackshear/frigate#14789] [Ask A Question] [answered]
Hi, a few times recently I've had issues with either no frames being received, or other issues like my NAS restarting which has stopped recordings until the addon is restarted. Is it possible to setup something perhaps in home assistant to alert me when no frames are received for one/more cameras, or even no new recordings have been made for a while? I asked ChatGPT and it came up with something but I don't trust it, so thought I'd ask here. I'll paste what it suggested incase there's something to it. Thanks in advance. I have the watchdog enabled in home assistant, but it doesn't catch these problems.

`alias: "Alert - Frigate Camera No Frames"
description: "Sends notification when Frigate detects no frames from cameras"
trigger:
  # Using a trigger pattern that works with multiple entities
  platform: event
  event_type: state_changed
  event_data:
    domain: sensor
  variables:
    is_frigate_fps: >
      {{ trigger.event.data.entity_id.startswith('sensor.frigate_') and 
         trigger.event.data.entity_id.endswith('_camera_fps') }}
    new_state: "{{ trigger.event.data.new_state.state }}"
    old_state: "{{ trigger.event.data.old_state.state }}"
condition:
  - condition: template
    value_template: "{{ is_frigate_fps }}"
  - condition: template
    value_template: "{{ new_state == '0' }}"
  - condition: template
    value_template: "{{ old_state != '0' }}"
  - condition: template
    value_template: "{{ trigger.event.data.new_state.state != 'unavailable' }}"
action:
  - service: notify.mobile_app_your_phone
    data:
      title: "⚠️ Camera Error Detected"
      message: >
        No frames receiving from camera: {{ trigger.event.data.entity_id.split('.')[1].split('_')[1] }}
        Last update: {{ trigger.event.data.new_state.last_changed }}
      data:
        push:
          sound: 
            name: "default"
            critical: 1
          priority: high
          ttl: 0
  
  # Optional: Send to Home Assistant persistent notifications
  - service: persistent_notification.create
    data:
      title: "Camera Error"
      message: >
        No frames receiving from camera: {{ trigger.event.data.entity_id.split('.')[1].split('_')[1] }}
        Last update: {{ trigger.event.data.new_state.last_changed }}

  # Optional: Log the error
  - service: system_log.write
    data:
      level: error
      message: "Frigate camera {{ trigger.event.data.entity_id.split('.')[1].split('_')[1] }} not receiving frames"

  # Optional: Wait and check if still at 0 FPS after delay
  - delay:
      minutes: 1
  - condition: template
    value_template: "{{ states(trigger.event.data.entity_id) == '0' }}"
  - service: notify.mobile_app_your_phone
    data:
      title: "⚠️ Camera Still Down"
      message: >
        Camera {{ trigger.event.data.entity_id.split('.')[1].split('_')[1] }} 
        has been down for over 1 minute`

**Comment by hawkeye217** [comment, marked as answer]
Not bad, ChatGPT, not bad.

You can use `sensor.<camera_name>_camera_fps` for a camera named `<camera_name>`, as it suggested. When the camera is working, the sensor value should be the `detect` fps you've defined in your config (default is 5). When the camera is down, this value will be 0.

**Reply by matknowles** [reply]
ok that's great, thanks for confirming

**Reply by Shiroe93** [reply]
I had the same issue but my frame didn't go to 0 but remained at a constant random value 
Like this (Red circle)
<img width="1080" height="698" alt="Screenshot_20251227_192142_Firefox" src="https://github.com/user-attachments/assets/3e7e9741-4ee3-402e-b5d1-56557fac23b2" />
Any other way to detect this in 0.16.3

**Reply by NickM-27** [reply]
You're showing detection fps. Not camera fps, which is the correct sensor to use

**Reply by Shiroe93** [reply]
Camera fps in the same time period is stuck at 5 fps no change of any kind that's the only anomaly i see 
Camera fps is like this even if the camera were not working from 16:51 of the 26 to 10:01 of the 27
![Screenshot_20251227_203245_Firefox.png](https://github.com/user-attachments/assets/32928209-c91b-4223-a1b0-4b9f3077be51)
The little blip you see is from me rebooting the add on

**Comment by matknowles** [comment]
for anyone who is interested in this, I have had to manually enable the sensor in the integration settings under diagnostic
